### DISCUSSION TITLE
enum/variant type in slint?

### DISCUSSION THREAD

**Discussion by anderslanglands** [slint-ui/slint#3080] [Q&A] [answered]
Is it possible to map a rust enum to slint? I have an application like a 3d model viewer where every object can have any number of arbitrarily typed, named properties, and I want to show those (and make them editable) to the user. 

I don't see anything like a sum type in the slint docs, so I'm currently maintaining one model per type, then another model stores type info per property and indices into each of the type-specific models to get the actual data to display. This works, but with ~20 types to handle it's going to get extremely unwieldy to maintain.

**Comment by ogoffart** [comment, marked as answer]
Right now there is no such things in the slint language.

There are some discussion in https://github.com/slint-ui/slint/issues/1307#issuecomment-1144759934

**Reply by anderslanglands** [reply]
OK thanks!
